neurokit2/complexity/utils.py

```diff
@@ -121,11 +121,43 @@ def _get_r(signal, r="default", dimension=2):
     https://github.com/CSchoel/nolds/blob/d8fb46c611a8d44bdcf21b6c83bc7e64238051a4/nolds/measures.py#L752
 
     """
-    if isinstance(r, str) or (r is None):
+    def _default_r(signal, dimension):
         constant = 0.11604738531196232
         r = constant * np.std(signal, ddof=1) * (0.5627 * np.log(dimension) + 1.3334)
+        return r
+
+    # r = "default"
+    if isinstance(r, str) or (r is None):
+        # Get different r values per channel and find mean
+        if signal.ndim > 1:
+            r_list = []
+            for i, col in enumerate(signal):
+                value = _default_r(signal[col], dimension=dimension)
+                r_list.append(value)
+            optimal_r = np.mean(r_list)
```

Comment:
For optimizing tolerance it's probably more elegant to have the `_get_r` utils function to incorporate the optimizing since this is shared across multiple entropy files - and then for other more unique parameters (such as those generated by `_fractal_correlation_get_r` and `_fractal_higuchi_optimal_k`) the optimizing can then be done in their respective complexity function files

for a visualization of the mean `r` in this case:
![Figure_1](https://user-images.githubusercontent.com/54053748/131950702-61fc4232-c755-40a4-bd76-4a79dba9b23a.png)

Reply:
mmmh

Reply:
> _get_r utils function to incorporate the optimizing since this is shared across multiple entropy file

yeah that looks good